tutorials/13_diagnosis_sbc.ipynb

```diff
@@ -0,0 +1,1897 @@
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\r",
+      " Training neural network. Epochs trained:  117"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\r",
+      " Training neural network. Epochs trained:  118"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\r",
+      " Training neural network. Epochs trained:  119"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\r",
+      " Training neural network. Epochs trained:  120"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\r",
+      " Training neural network. Epochs trained:  121"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\r",
+      " Training neural network. Epochs trained:  122"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\r",
+      " Training neural network. Epochs trained:  123"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\r",
+      " Training neural network. Epochs trained:  124"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\r",
+      " Training neural network. Epochs trained:  125"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\r",
+      " Training neural network. Epochs trained:  126"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\r",
+      " Training neural network. Epochs trained:  127Neural network successfully converged after 127 epochs.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from sbi.utils.get_nn_models import posterior_nn\n",
+    "\n",
+    "#we use a mdn model to have a fast turnaround with training\n",
+    "defn = posterior_nn(model='mdn')\n",
+    "inferer = SNPE(prior, density_estimator=defn)\n",
+    "\n",
+    "# let's not save the density_estimator in order to do NPE\n",
+    "# (we have enough simulations)\n",
+    "_ = inferer.append_simulations(theta, x).train()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "d10bd3a18b3f458e9cbcdeb203b3e633",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Drawing 15000 posterior samples:   0%|          | 0/15000 [00:00<?, ?it/s]"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "posterior = inferer.build_posterior()\n",
+    "posterior_samples = posterior.sample((15_000,), x = x_o).detach()\n",
```

Comment:
samples should be detached already, no? If not this is a bug and we should create an issue.

Reply:
well spotted ... thanks.
315c59e